pkg/slack/filter.go

```diff
@@ -0,0 +1,84 @@
+package slack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/oursky/github-actions-manager/pkg/github/jobs"
+	"k8s.io/utils/strings/slices"
+)
+
+type messageFilterRule struct {
+	conclusions []string
+	// branches    []string
+	workflows []string
+}
+type MessageFilter struct {
+	whitelists []messageFilterRule
+}
+
+func (rule messageFilterRule) Pass(run *jobs.WorkflowRun) bool {
+	if len(rule.conclusions) > 0 && !slices.Contains(rule.conclusions, run.Conclusion) {
+		return false
+	}
+	if len(rule.workflows) > 0 && !slices.Contains(rule.workflows, run.Name) {
+		return false
+	}
+	return true
+}
+
+func (mf MessageFilter) Any(run *jobs.WorkflowRun) bool {
+	if len(mf.whitelists) == 0 {
+		return true
+	}
```

Comment:
If empty, should return `false` by the definition

Reference https://en.wikipedia.org/wiki/Existential_quantification#The_empty_set